pkg/processor/trigger/kafka/trigger.go

```diff
@@ -404,6 +404,13 @@ func (k *kafka) eventSubmitter(claim sarama.ConsumerGroupClaim, submittedEventCh
 func (k *kafka) cancelEventHandling(workerInstance *worker.Worker,
 	claim sarama.ConsumerGroupClaim) error {
 	if workerInstance.SupportsRestart() {
+
+		k.Logger.WarnWith("Cancelling event handling",
+			"topic", claim.Topic(),
+			"partition", claim.Partition())
+		if err := workerInstance.Drain(); err != nil {
```

Comment:
Not sure why this is needed here too, as the worker is already drained earlier in `SignalWorkerDraining`